src/player.js

```diff
@@ -1670,12 +1692,13 @@ export default class Player extends FakeEventTarget {
   _selectEngineByPriority(): boolean {
     const streamPriority = this._config.playback.streamPriority;
     const preferNative = this._config.playback.preferNative;
-    const sources = this._config.sources;
+    const sources = this._sources;
     for (let priority of streamPriority) {
       const engineId = typeof priority.engine === 'string' ? priority.engine.toLowerCase() : '';
       const format = typeof priority.format === 'string' ? priority.format.toLowerCase() : '';
       const Engine = EngineProvider.getEngines().find(Engine => Engine.id === engineId);
       if (Engine) {
+        // $FlowFixMe
```

Comment:
same

Reply:
Same answer here, I changed everything to typed objects